pkg/cmd/roachtest/tests/restore.go

```diff
@@ -414,23 +414,130 @@ func (tpccIncData) runRestoreDetached(
+// this test.
+//
+// 2) Setup the cluster to run tpc-e init with 500k customers, using
+// the repro steps in the appendix of Nathan's v22.2
+// Scalability & Efficiency Evaluation using TPC-E doc
+// https://docs.google.com/document/d/1wzkBXaA3Ap_daMV1oY1AhQqlnAjO3pIVLZTXY53m0Xk/edit
+//
+// 3) Set the gc ttl for the tpce database to 25 hrs
+//
+// 4) Create a backup schedule with revision history and hourly incremental backups.
+//
+// 6) Run the tpce workload for 24 hours with 100k active customers.
+type tpce10TB struct{}
+
+func (tpce10TB) name() string {
+	return "TPCE10TB"
+}
+
+func (tpce10TB) runRestore(ctx context.Context, c cluster.Cluster) {
+	// Restore from the first full backup AOST in the last incremental backup in the chain.
+	c.Run(ctx, c.Node(1), `./cockroach sql --insecure -e "
+				RESTORE DATABASE tpce FROM '/2022/11/06-124208.52' IN
+				'gs://cockroach-fixtures/backups/tpc-e/rev-history=true,inc-count=23,cluster/customers=500k/22.1.8?AUTH=implicit'
+				AS OF SYSTEM TIME '2022-11-06 23:40:22'"`)
+}
+
+func (tpce10TB) runRestoreDetached(
+	ctx context.Context, t test.Test, c cluster.Cluster,
+) (jobspb.JobID, error) {
+	// Restore from the first full backup AOST in the last incremental backup in the chain.
+	c.Run(ctx, c.Node(1), `./cockroach sql --insecure -e "
+				RESTORE DATABASE tpce FROM '/2022/11/06-124208.52' IN
+				'gs://cockroach-fixtures/backups/tpc-e/rev-history=true,inc-count=23,cluster/customers=500k/22.1.8?AUTH=implicit'
+				AS OF SYSTEM TIME '2022-11-06 23:40:22' WITH detached"`)
+	db, err := c.ConnE(ctx, t.L(), c.Node(1)[0])
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to connect to node 1; running restore detached")
+	}
+
+	var jobID jobspb.JobID
+	if err := db.QueryRow(`SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'RESTORE' ORDER BY created DESC LIMIT 1`).Scan(
+		&jobID); err != nil {
+		return 0, err
+	}
+
+	return jobID, nil
+}
+
 var _ testDataSet = tpccIncData{}
 
+// checkDetachedRestore returns when the detached restore has completed
+func checkDetachedRestore(
+	ctx context.Context, t test.Test, c cluster.Cluster, jobID jobspb.JobID,
+) error {
+	succeededJobTick := time.NewTicker(time.Minute * 1)
+	defer succeededJobTick.Stop()
+	done := ctx.Done()
+	for {
+		select {
+		case <-done:
+			return ctx.Err()
+		case <-succeededJobTick.C:
+			// Open a new connection for every check to prevent connection flakes.
+			conn, err := c.ConnE(ctx, t.L(), c.Node(1)[0])
+			if err != nil {
+				return errors.Wrapf(err, "failed to open connection to sql server")
+			}
+
+			var status string
+			if err = conn.QueryRow(`SELECT status FROM [SHOW JOBS] WHERE job_id = $1`, jobID).Scan(
+				&status); err != nil {
+				return errors.Wrapf(err, "failed to check restore job status")
+			}
+			if status == string(jobs.StatusSucceeded) {
+				return nil
+			} else if status != string(jobs.StatusRunning) {
+				return errors.Newf("job unexpectedly found in %s state", status)
```

Comment:
Now that we are running with detached it might become harder to see the actual reason for the job failure since it will not be returned to the client. Should we be doing something special if we find the job in a reverting or failed state? Printing just the status might not be as useful as also printing the error we stash in the job? wdyt

Reply:
Done.